### DISCUSSION TITLE
Publish the `node-resolver-rs` create to crates.io?

### DISCUSSION THREAD

**Discussion by tryggvigy** [parcel-bundler/parcel#8923] [Ideas] [answered]
Hello! Parcels' `node-resolver-rs` seems to be the most complete implementation of a node resolver in rust at the moment. 

Unless I'm missing something, it doesn't seem to be published to crates.io. I think it could be very useful for the rust community! 

For example
- swc has it's own resolver which doesn't implement "imports" and "exports" keys yet.
- Any rust programs doing dependency analysis/bundling/etc.

I'm working on a rust program to check a project's ESM readiness. I've been using swc node resolver but since it doesn't support "exports" my script is misclassifying certain transitive dependencies as CJS when they do in fact export ESM. I think this resolver is exactly what I need

**Comment by mischnic** [comment, marked as answer]
@devongovett wrote in https://discord.com/channels/894288336095690753/894668995444219914/1085044705827500045 :

> I plan to publish the resolver crate, yes. Want to audit the API first. Should be able to do that soon now that the implementation has mostly stabilized

In the meantime, you should be able to use it via the git repo:
```toml
[dependencies]
parcel-resolver = { git = "https://github.com/parcel-bundler/parcel.git" }
```
